dio/lib/src/adapters/browser_adapter.dart

```diff
@@ -111,15 +114,17 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
     // so we can check it beforehand.
     if (requestStream != null) {
       if (connectTimeoutTimer != null) {
-        xhr.upload.onProgress.listen((event) {
+        void cancelConnectTimeoutTimer(web.ProgressEvent _) {
           connectTimeoutTimer?.cancel();
           connectTimeoutTimer = null;
-        });
+        }
+
+        xhr.upload.onprogress = cancelConnectTimeoutTimer.toJS;
```

Comment:
Is there a way to pass these functions anonymously?

Reply:
Yes, we can write like `xhr.upload.onprogress = (web.ProgressEvent event){}.toJS` this.